rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -7949,3 +7949,50 @@ fn check_install_code_in_wasm64_mode_is_charged_correctly() {
     assert_lt!(balance64, balance32);
     assert_lt!(execution_cost32, execution_cost64);
 }
+
+#[test]
+fn subnet_info_canister_call_succeeds() {
+    let own_subnet_id = subnet_test_id(1);
+    let mut test = ExecutionTestBuilder::new()
+        .with_own_subnet_id(own_subnet_id)
+        .build();
+    let uni_canister = test
+        .universal_canister_with_cycles(Cycles::new(1_000_000_000_000))
+        .unwrap();
+    let payload = SubnetInfoArgs {
+        subnet_id: own_subnet_id.get(),
+    }
+    .encode();
+    let uc_call = wasm()
+        .call_simple(
+            CanisterId::ic_00(),
+            Method::SubnetInfo,
+            call_args().other_side(payload),
+        )
+        .build();
+    let result = test.ingress(uni_canister, "update", uc_call).unwrap();
+    let bytes = match result {
+        WasmResult::Reply(bytes) => bytes,
+        WasmResult::Reject(err_msg) => panic!("Unexpected reject, expected reply: {}", err_msg),
+    };
+    let SubnetInfoResponse { replica_version } = Decode!(&bytes, SubnetInfoResponse).unwrap();
+    assert!(!replica_version.is_empty());
+}
+
+#[test]
```

Comment:
Maybe one more test where you try sending a request with the wrong `subnet_id`?

Reply:
As discussed afk, this can be a follow-up SM-test.